packages/remix-server-runtime/cookies.ts

```diff
@@ -99,12 +99,17 @@ export function createCookie(
     },
     async parse(cookieHeader, parseOptions) {
       if (!cookieHeader) return null;
-      let cookies = parse(cookieHeader, { ...options, ...parseOptions });
-      return name in cookies
-        ? cookies[name] === ""
-          ? ""
-          : await decodeCookieValue(cookies[name], secrets)
-        : null;
+      const cookies = parse(cookieHeader, { ...options, ...parseOptions });
+
+      if (name in cookies) {
+        if (cookies[name] === "") {
+          return "";
+        } else {
+          return decodeCookieValue<Value>(cookies[name], secrets);
+        }
```

Comment:
is the omission of `await` in front of `decodeCookieValue` intentional?

Reply:
I removed `await` since it wasn't necessary; in either case, the function will return a promise.